### DISCUSSION TITLE
Vim mod not convenient?

### DISCUSSION THREAD

**Discussion by dupeiran001** [zed-industries/zed#6624] [Feature Requests]
having enabled vim mode by `"vim_mode": true`
But it only simulate some basic functions like cursor movement. There are lots of features to be add

- any command plate like `:w` don't work
- visual block not working like `^v`
- dd multiple lines not working, like '3dd'
- finding next not working, like `fa` and `n`, it should jump to next 'a'
- etc

Vim mode is quite important for me, so is it possibly be fixed by editing key map? 
Or is it scheduled in some future version?

**Comment by Warkanlock** [comment]
Adding to the discussion, ``gd`` (go-to-def) activate by itself the visual selection mode, which it's kinda strange

**Reply by anttti** [reply]
Yes! This is actually a bit distracting. Also `*`, `(` and `)` do not seem to be implemented (yet?).

**Reply by thraizz** [reply]
Jumping from bracket to bracket with % does not work either, but I cannot remember if that is actually non-native vim functionality

**Comment by torfason** [comment]
Adding one more:

- `P` (capital P for pasting line above) does not seem to be working for me.

**Comment by alfpedersen** [comment]
Adding one more again:

- `J` (capital J for joining line below cursor with line of cursor) is not working.

**Comment by everdimension** [comment]
Some more:

* <kbd>s</kbd> doesn't work (should delete under selection and go to insert mode)
* <kbd>ctrl+d</kbd> works in normal mode, but works incorrectly when there's a selection. It should scroll down and move the selection, instead, it deletes text under selection
* This one might be a "wontfix", but: Start search using <kbd>/</kbd>, enter characters, press enter. After this, pressing <kbd>n</kbd> should search next word, instead it just enters the "n" character into search input
* Tree view should be navigatable with <kbd>j</kbd> and <kbd>k</kbd>. But right now at least it's navigatable with <kbd>ctrl+p</kbd> and <kbd>ctrl+n</kbd>, cool!
* <kbd>shift+[</kbd>, <kbd>shift+]</kbd> do not work ("go to next paragraph")
* <kbd>g;</kbd> doesn't work (go to previous edit)


And some more, not specific to vim, but to keyboard-only users:

* Variable types and linting warnings should be revealable with a keyboard shortcut, e.g. <kbd>shift+k</kbd>, and not only by hovering the mouse over the token.
* I didn't find how to focus tree view from keyboard
* Shortcuts for actions like "go to definition" should be configurable
  * But even if not, the context menu that is currently opened by "right click" should be opened with a keyboard shortcut as well
* Some more shortcuts are needed, which I maybe did not find, like
  * Go to next/prev linting / ast error in file
  * Go to next/prev git-new/git-modified line in file

---

So far, the editor feels awesome! Thank you so much for hard work, I hope all of these features will make it to the beta soon

**Reply by dupeiran001** [reply]
Thanks for your appendix, that’s much detailed

**Comment by dcow** [comment]
Text object navigation doesn't seem to work either `[,],{,}, etc.`

**Comment by franck-boisseron** [comment]
Hi, 

I really like the slickness of this editor and so far the feeling is very good, but I must admit that the vim_mode could need some work.. so many native vim feature are missing that it's barely usable, let not even think about plugin. The remap to `jk` works but prevent typing j in insert mode. You can't use `: command`, I'm used to saving by typing `:wa` or `:reg` to list register, `:vsplit` and so on. Searching for occurence either by `/` or `f` or `t` does not work, or in case of `/` requires using the mouse.

To add to the list

`Shift-O` in visual mode does not go to the other side of the selection.
`R` should enable replace word in place, but does nothing
`%`  inside brace /bracket.. should go to the other brace
`Ctrl-i` `Ctrl-o` should go to previous/next location
`v+i+b` visual selection inside or at block level does not work
`Ctrl+w+(hjkl)` does not switch window
Having a colored status bar to check in which mode we are could be useful.

**Reply by franck-boisseron** [reply]
using either `c+i+t` or `d+i+t` inside an html file, does not edit or delete the value inside the tag, and do not bring you to the nearest tag.

**Reply by franck-boisseron** [reply]
using `Ctrl-r` + `"` in insert mode does not paste the value inside the `"` register. Actually I'm not sure there is a notion of register at all.

**Reply by franck-boisseron** [reply]
`Shift` + 'h' or 'l' or 'm' does not make the cursor jump to top of screen, middle and low part
`<` and `>` for indentation do not work in visual mode

I think I'll simply wait for an update on the `vim_mode`. I know that few commands can be implemented by using the key bidding settings (already did for emulate the ladder key as space and use it for splitting window) but it's quite a pain.

I really do like this editor especially for it support and comfort when writing Rust, but I really hope that the `vim_mode` get a big update with at least the main basic `vi` feature, because so far it may better be called `vim_motion` than `vim_mode`.

**Comment by everdimension** [comment]
* Vim marks do not work
* `''` to jump to previous location doesn't work

**Comment by dunkbing** [comment]
Shift + k in insert mode does not make a capital K.

**Reply by torfason** [reply]
This last one definitely does not reproduce for me – could this be some key mapping that is corrupting this?

**Reply by everdimension** [reply]
I discovered I had the same issue and it was because of a custom mapping. 
What I had was:
```
  {
    "context": "Editor",
    "bindings": {
      ...,
      "shift-k": "editor::Hover"
    }
  },
```
 To fix, I moved `shift-k` binding to a different context:

```
  {
    "context": "Editor && vim_mode != insert",
    "bindings": {
      "shift-k": "editor::Hover"
    }
  },
  {
```

**Comment by everdimension** [comment]
Does anyone know if the issues in this discussion are something the team is working on or at least acknowledges?

**Reply by hovsater** [reply]
We're aware that there's several short comings in the Vim emulation right now. Work on the Vim emulation layer is currently on hold as our priorities have changed. With that said, there should be some exciting news for you all in the near future. 🙂

**Reply by everdimension** [reply]
Ok, can't wait! Thx

**Comment by JosephTLyons** [comment]
Just wanted to mention that we are working on vim mode now and there *should* be a stream of improvements coming in through each update.

**Reply by everdimension** [reply]
That's great news! Can't wait to be able to switch to his editor

**Comment by QuentinWidlocher** [comment]
Also, not directly related to VIM mode but the lack of relative line numbers makes it harder to work with.

See #998

**Comment by txtyash** [comment]
Vim emulation in vs code and zed will never suffice for vim/neovim users

**Reply by thraizz** [reply]
But vscode emulation is really close 😸

**Reply by txtyash** [reply]
you can't even use `:g` :shrug:

**Reply by distek** [reply]
_Two years later_

`:g` works for me now!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
